src/main/java/com/minecolonies/coremod/commands/colonycommands/ColonyTeleportCommand.java

```diff
@@ -57,26 +74,91 @@ public boolean canRankUseCommand(@NotNull final Colony colony, @NotNull final En
     }
 
     @Override
-    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
     {
-        //see if player is allowed to use in the configs
-        if (sender instanceof EntityPlayer && args.length == 1)
+        Colony colony = actionMenu.getColonyForArgument("colony");
+        if (null == colony)
         {
-            final int colonyID = getIthArgument(args, 0, -1);
-            if (colonyID != -1 && canPlayerUseCommand((EntityPlayer) sender, COLONYTP, colonyID))
+            final EntityPlayer player = actionMenu.getPlayerForArgument("player");
+            if (player != null)
             {
-                final Colony colonyIn = ColonyManager.getColony(((EntityPlayer) sender).world, sender.getPosition());
-                if (isPlayerOpped(sender)
-                        || (colonyIn != null
-                        && colonyIn.hasTownHall()
-                        && colonyIn.getPermissions().hasPermission((EntityPlayer) sender, Action.TELEPORT_TO_COLONY)
-                        && ((EntityPlayer) sender).getDistanceSq(colonyIn.getBuildingManager().getTownHall().getLocation()) < MIN_DISTANCE_TO_TH))
+                IColony iColony = ColonyManager.getIColonyByOwner(server.getEntityWorld(), player);
+                if (null == iColony)
+                {
+                    if (sender instanceof EntityPlayer)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)